Use jreleaser, remove release-drafter.

netomi opened the pull request:
This fixes #263 .

The release workflow has been updated to use the latest version of jreleaser.
Additional changes:

 - remove release-drafter
 - add custom action to bump version for next development cylce

aalmiray commented:
The `BINARY` distribution type is supposed to be used for platform specific executables. If the project provides a Java based binary then the `JAVA_BINARY` distribution type is preferred, which is also the default value. 

https://jreleaser.org/guide/latest/concepts/distributions/binary.html

https://jreleaser.org/guide/latest/concepts/distributions/java-binary.html

The choice of distribution type determines which other features may be used later, such as package managers.

netomi commented:
marking as draft, doing some tests.

netomi commented:
Finally got it working: https://github.com/netomi/macos-notarization-service/actions/runs/7144907502

Need to cleanup the PR and squash commits.

netomi commented:
cleaned up PR, added additional checks for the release workflow:

- check if a tag with the same name already exists
- only allow releases from the main branch

netomi commented:
An example run of the release workflow in my fork: https://github.com/netomi/macos-notarization-service/actions/runs/7153981992

Resulting in this release: https://github.com/netomi/macos-notarization-service/releases/tag/v1.3.0
